grakn-kb/src/main/java/ai/grakn/kb/internal/concept/RelationshipEdge.java

```diff
@@ -57,21 +57,21 @@
     private final Logger LOG = LoggerFactory.getLogger(RelationshipEdge.class);
     private final EdgeElement edgeElement;
 
-    private final Cache<RelationshipType> relationType = new Cache<>(Cacheable.concept(), () ->
+    private final Cache<RelationshipType> relationType = Cache.createTxCache(Cacheable.concept(), () ->
```

Comment:
Maybe we can make this work like `Concept` by adding a `CacheOwner` interface? `Concept` and `RelationshipStructure` can implement it - then we don't need the factory method that doesn't take an owner.